compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -18136,11 +18176,11 @@ private STNode mergeQualifiedNameWithTypeDesc(STNode qualifiedName, STNode typeD
             case UNION_TYPE_DESC:
                 STUnionTypeDescriptorNode unionTypeDesc = (STUnionTypeDescriptorNode) typeDesc;
                 STNode newlhsType = mergeQualifiedNameWithTypeDesc(qualifiedName, unionTypeDesc.leftTypeDesc);
-                return createUnionTypeDesc(newlhsType, unionTypeDesc.pipeToken, unionTypeDesc.rightTypeDesc);
+                return mergeLhsTypeAndRhsType(newlhsType, unionTypeDesc.pipeToken, unionTypeDesc.rightTypeDesc);
```

Comment:
We can directly call `mergeLhsTypeAndRhsTypeWithUnion` right?. Also lets change the name of `mergeLhsTypeAndRhsTypeWithUnion` to `mergeAsUnionType`. Let's update the other places as well